### PR TITLE
Add testRender with CoroutineScope for SessionWorkflow

### LINKED ISSUE #1138: SessionWorkflow throws when trying to unit test it with RenderTester#testRender
When trying to unit test a Workflow that inherits from `SessionWorkflow` using `RenderTester#testRender` I get this error:

> SessionWorkflow should never call initialState without the CoroutineScope.
java.lang.IllegalStateException: SessionWorkflow should never call initialState without the CoroutineScope.
	at com.squareup.workflow1.SessionWorkflow.initialState(SessionWorkflow.kt:82)
	at com.squareup.workflow1.testing.RenderTesterKt.testRender(RenderTester.kt:29)
	

It looks like `RenderTester#testRender` is calling an overloaded `initialState` method that `SessionWorkflow` doesn't support, see https://github.com/square/workflow-kotlin/blob/b7e37bc04fff285fbb346c28c1008817a5a1578e/workflow-core/src/commonMain/kotlin/com/squareup/workflow1/SessionWorkflow.kt#L78-L84.

### PR DESCRIPTION
**Pull request by steve-the-edwards**
Fixes #1138

### CODE REVIEW

File: workflow-testing/src/main/java/com/squareup/workflow1/testing/RenderTester.kt [hunk 1]
```diff
@@ -19,14 +22,43 @@ import kotlin.reflect.KTypeProjection
...
 @Suppress("UNCHECKED_CAST")
 public fun <PropsT, OutputT, RenderingT> Workflow<PropsT, OutputT, RenderingT>.testRender(
   props: PropsT
 ): RenderTester<PropsT, *, OutputT, RenderingT> {
   val statefulWorkflow = asStatefulWorkflow() as StatefulWorkflow<PropsT, Any?, OutputT, RenderingT>
   return statefulWorkflow.testRender(
     props = props,
-    initialState = statefulWorkflow.initialState(props, null)
+    initialState = run {
+      require(this !is SessionWorkflow<PropsT, *, OutputT, RenderingT>) {
```
Comment (rjrjr) [inline workflow-testing/src/main/java/com/squareup/workflow1/testing/RenderTester.kt:34]: nice

File: workflow-testing/src/main/java/com/squareup/workflow1/testing/RenderTester.kt [hunk 2]
```diff
@@ -19,14 +22,43 @@ import kotlin.reflect.KTypeProjection
...
+
+/**
+ * Create a [RenderTester] to unit test an individual render pass of this [SessionWorkflow],
+ * using the workflow's [initial state][StatefulWorkflow.initialState], in the [workflowScope].
+ *
+ * See [RenderTester] for usage documentation.
+ */
+@OptIn(WorkflowExperimentalApi::class)
+@Suppress("UNCHECKED_CAST")
+public fun <PropsT, OutputT, RenderingT> Workflow<PropsT, OutputT, RenderingT>.testRender(
```
Comment (rjrjr) [inline workflow-testing/src/main/java/com/squareup/workflow1/testing/RenderTester.kt:50]: Can't you make this version an extension on `SessionWorkflow` and avoid the runtime check?

**Review by rjrjr — APPROVED** [review]

File: workflow-testing/src/main/java/com/squareup/workflow1/testing/RenderTester.kt (hunk 2 above)
Comment (steve-the-edwards) [reply workflow-testing/src/main/java/com/squareup/workflow1/testing/RenderTester.kt:50]: ya I thought about that but then I thought about how in your tests you might not have the reified `SessionWorkflow` type? Maybe you were passing it around or dealign with it just as a `Workflow`? Maybe I was overthinking it though. Perhaps we could just make it an extension on `SessionWorkflow` 🤔

Comment (rjrjr) [reply workflow-testing/src/main/java/com/squareup/workflow1/testing/RenderTester.kt:50]: I think so. I'm going to write a unit test against `class RealFooWorkflow`, not `interface FooWorkflow`.

Comment (steve-the-edwards) [reply workflow-testing/src/main/java/com/squareup/workflow1/testing/RenderTester.kt:50]: ya, I think you are right. I will change it.

- [ ] testRender + CoroutineScope extension extends `SessionWorkflow` not `Workflow`

**Pull request merged**